File: backend/schemas/journal.py
from pydantic import BaseModel
from datetime import date, datetime
from typing import Optional


class JournalEntryCreate(BaseModel):
    entry_date: date
    title: Optional[str] = None
    body: str = ""


class JournalEntryUpdate(BaseModel):
    title: Optional[str] = None
    body: Optional[str] = None
    entry_date: Optional[date] = None


class JournalEntryResponse(BaseModel):
    id: str
    user_id: str
    entry_date: date
    title: Optional[str]
    body: str
    created_at: datetime
    updated_at: datetime
    primary_sentiment: Optional[str] = None
    sentiment_score: Optional[float] = None
    energy_level: Optional[str] = None
    key_themes: Optional[list[str]] = None
    one_line_summary: Optional[str] = None
    keywords: Optional[list[str]] = None
    analysis_status: Optional[str] = None
    analysed_at: Optional[datetime] = None

    class Config:
        from_attributes = True


class JournalListResponse(BaseModel):
    items: list[JournalEntryResponse]
    total: int
    page: int
    has_next: bool

File: frontend/src/types/journal.ts
export interface JournalEntry {
  id: string
  user_id: string
  entry_date: string
  title: string | null
  body: string
  created_at: string
  updated_at: string
  // AI analysis fields
  primary_sentiment?: string | null
  sentiment_score?: number | null
  energy_level?: string | null
  key_themes?: string[] | null
  one_line_summary?: string | null
  keywords?: string[] | null
  analysis_status?: string | null
  analysed_at?: string | null
}

export interface JournalListResponse {
  items: JournalEntry[]
  total: number
  page: number
  has_next: boolean
}

export interface CreateJournalEntry {
  entry_date: string
  title?: string
  body?: string
}

export interface UpdateJournalEntry {
  title?: string
  body?: string
  entry_date?: string
}
